File: Lecture1/exercise/CMYKtoRGB.java
public class CMYKtoRGB{
    public static void main(String[] args){
        double c = Double.parseDouble(args[0]);
        double m = Double.parseDouble(args[1]);
        double y = Double.parseDouble(args[2]);
        double k = Double.parseDouble(args[3]);

        double w = 1 - k;
        double r = 255 * w * ( 1 - c );
        double g = 255 * w * ( 1 - m );
        double b = 255 * w * ( 1 - y );

        System.out.println("red   = " + (int) Math.round(r));
        System.out.println("green = " + (int) Math.round(g));
        System.out.println("blue  = " + (int) Math.round(b));
    }
}

File: Lecture1/c_repro/CMYKtoRGB.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

void main(int argc, char* argv[]){
    float c = atof(argv[1]);
    float m = atof(argv[2]);
    float y = atof(argv[3]);
    float k = atof(argv[4]);

    float w = 1 - k;
    float r = 255 * w * ( 1 - c );
    float g = 255 * w * ( 1 - m );
    float b = 255 * w * ( 1 - y );

    printf("red   = %d.\n", (int) roundf(r));
    printf("green = %d.\n", (int) roundf(g));
    printf("blue  = %d.\n", (int) roundf(b));
}